File: backend/app/schemas/graph.py
from uuid import UUID

from pydantic import BaseModel


class GraphNode(BaseModel):
    id: UUID
    name: str
    type: str
    mastery: int
    importance: int
    knowledge_gap: bool = False


class GraphEdge(BaseModel):
    id: UUID
    source: UUID
    target: UUID
    type: str


class GraphRead(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]
    truncated: bool = False


class ConceptSummary(BaseModel):
    id: UUID
    name: str
    type: str


class ConceptSourceSummary(BaseModel):
    id: UUID
    title: str
    type: str


class ConceptDetail(BaseModel):
    id: UUID
    name: str
    type: str
    description: str
    importance: int
    confidence: int
    mastery: int
    knowledge_status: str
    aliases: list[str]
    prerequisites: list[ConceptSummary]
    unlocks: list[ConceptSummary]
    sources: list[ConceptSourceSummary]

File: frontend/src/shared/types/graph.ts
export type GraphNode = {
  id: string;
  name: string;
  type: string;
  mastery: number;
  importance: number;
  knowledge_gap: boolean;
};

export type GraphEdge = {
  id: string;
  source: string;
  target: string;
  type: string;
};

export type KnowledgeGraph = {
  nodes: GraphNode[];
  edges: GraphEdge[];
  truncated: boolean;
};

export type ConceptSummary = {
  id: string;
  name: string;
  type: string;
};

export type ConceptDetail = {
  id: string;
  name: string;
  type: string;
  description: string;
  importance: number;
  confidence: number;
  mastery: number;
  knowledge_status: string;
  aliases: string[];
  prerequisites: ConceptSummary[];
  unlocks: ConceptSummary[];
  sources: Array<{ id: string; title: string; type: string }>;
};
